internal/resource/http/http_test.go

```diff
@@ -311,6 +312,88 @@ func createResource(t *testing.T, artistHandler resourceHandler) {
 	require.Equal(t, http.StatusOK, rsp.Result().StatusCode)
 }
 
+func TestResourceReadHandler(t *testing.T) {
+	aclResolver := &resourceSvc.MockACLResolver{}
+	aclResolver.On("ResolveTokenAndDefaultMeta", testACLTokenArtistReadPolicy, mock.Anything, mock.Anything).
+		Return(svctest.AuthorizerFrom(t, demo.ArtistV1ReadPolicy, demo.ArtistV2ReadPolicy), nil)
+	aclResolver.On("ResolveTokenAndDefaultMeta", testACLTokenArtistWritePolicy, mock.Anything, mock.Anything).
+		Return(svctest.AuthorizerFrom(t, demo.ArtistV1WritePolicy, demo.ArtistV2WritePolicy), nil)
+	aclResolver.On("ResolveTokenAndDefaultMeta", fakeToken, mock.Anything, mock.Anything).
+		Return(svctest.AuthorizerFrom(t, ""), nil)
+
+	client := svctest.RunResourceServiceWithACL(t, aclResolver, demo.RegisterTypes)
+
+	v2ArtistHandler := resourceHandler{
+		resource.Registration{
+			Type:  demo.TypeV2Artist,
+			Proto: &pbdemov2.Artist{},
+		},
+		client,
+		parseToken,
+		hclog.NewNullLogger(),
+	}
```

Comment:
I think it is best to test this from the `NewHandler` entry point so that we are testing the registration flow as well. To achieve that you need:
```
r := resource.NewRegistry()
	demo.RegisterTypes(r)

	handler := NewHandler(client, r, parseToken, hclog.NewNullLogger())
```

Reply:
right, better way to handle this! thanks!